apps/sms/services/js/device_storage/device_storage_shim.js

```diff
@@ -0,0 +1,59 @@
+/* global BridgeServiceMixin,
+          BroadcastChannel
+*/
+
+/* exported DeviceStorageShim */
+
+(function(exports) {
+'use strict';
+
+/**
+ * Name of the service for device storage API shim.
+ * @type {string}
+ */
+const SERVICE_NAME = 'device-storage-shim';
+
+const EVENTS = ['change'];
+
+const METHODS = ['lowDiskSpace'];
+
+var deviceStorage = null;
+
+var DeviceStorageShim = {
+  init(appInstanceId, realDeviceStorage) {
+    if (!realDeviceStorage) {
+      console.error('Invalid deviceStorage for shim initialization');
+      return;
+    }
+
+    deviceStorage = realDeviceStorage;
+
+    EVENTS.forEach((event) => {
+      deviceStorage.addEventListener(event, (evt) => {
+        // Workaround to avoid the duplicated events by inserting
+        // appInstanceId into event name in bridge mixin.
+        evt.appInstanceId = appInstanceId;
```

Comment:
I think you are right that we don't need the workaround here. I was confused by the message event broadcasting case, but it's because the messaging client and service was sharing with same channel. For the storage client - shim case, it already had unique channel and we don't need to worry about receiving the event from other instance.